app/components/verkiezingen/warning-amount-mandatarissen-for-orgaan-alert.js

```diff
@@ -62,60 +78,59 @@ export default class VerkiezingenWarningAmountMandatarissenForOrgaanAlertCompone
           const totalForMandaat = this.args.mandatarissen.filter((m) => {
             return m.get('bekleedt.id') === key;
           }).length;
-          let message = null;
-          let messagePositionInArray = null;
           if (totalForMandaat > value.max) {
-            message = `Teveel mandaten gevonden voor "${value.label}". ${totalForMandaat} van maximum ${value.max}`;
-            messagePositionInArray = this.warningMessages.length + 1;
+            const message = `Teveel mandaten gevonden voor "${value.label}". (${totalForMandaat}/${value.max})`;
+            this.warningMessages.pushObject({
+              message: message,
+              id: this.createAlertId(value.mandaatId, this.errorMessageMaxId),
+            });
           }
           if (totalForMandaat < value.min) {
-            message = `Te weinig mandaten gevonden voor "${value.label}". (${totalForMandaat}/${value.min})`;
-            messagePositionInArray = this.warningMessages.length + 1;
-          }
-
-          this.setMessageForMandaat(key, message, messagePositionInArray);
-          if (message) {
+            const message = `Te weinig mandaten gevonden voor "${value.label}". (${totalForMandaat}/${value.min})`;
             this.warningMessages.pushObject({
               message: message,
-              position: messagePositionInArray,
+              id: this.createAlertId(value.mandaatId, this.errorMessageMinId),
             });
           }
         }
       )
     );
+    await this.onUpdate();
   }
 
-  setMessageForMandaat(mandaatId, message, messagePositionInArray) {
-    if (!mandaatId || !messagePositionInArray) {
-      return;
-    }
-
-    const current = this.mandaatValueMapping.get(mandaatId);
-    delete current.message;
-    this.mandaatValueMapping.set(mandaatId, {
-      ...current,
-      message,
-    });
+  setPossibleWarningIds() {
```

Comment:
Created a getter for it first, but now it is only updated when the mapping is updated so should be less loops